NightscoutServiceKit/Extensions/StoredDosingDecision.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  StoredDosingDecision.swift
+//  NightscoutServiceKit
+//
+//  Created by Darin Krauss on 10/17/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import LoopKit
+import NightscoutUploadKit
+
+extension StoredDosingDecision {
+    
+    var iobStatus: IOBStatus? {
+        guard let insulinOnBoard = insulinOnBoard else {
+            return nil
+        }
+        return IOBStatus(timestamp: insulinOnBoard.startDate, iob: insulinOnBoard.value)
+    }
+    
+    var cobStatus: COBStatus? {
+        guard let carbsOnBoard = carbsOnBoard else {
+            return nil
+        }
+        return COBStatus(cob: carbsOnBoard.quantity.doubleValue(for: HKUnit.gram()), timestamp: carbsOnBoard.startDate)
+    }
+    
+    var predictedBG: PredictedBG? {
+        guard let predictedGlucose = predictedGlucose, let startDate = predictedGlucose.first?.startDate else {
+            return nil
+        }
+        return PredictedBG(startDate: startDate, values: predictedGlucose.map { $0.quantity })
+    }
+    
+    var recommendTempBasal: RecommendedTempBasal? {
+        guard let tempBasalRecommendationDate = tempBasalRecommendationDate else {
+            return nil
+        }
+        return RecommendedTempBasal(timestamp: tempBasalRecommendationDate.date,
+                                    rate: tempBasalRecommendationDate.recommendation.unitsPerHour,
+                                    duration: tempBasalRecommendationDate.recommendation.duration)
+    }
+    
+    var loopEnacted: LoopEnacted? {
+        guard case .some(.tempBasal(let tempBasal)) = pumpManagerStatus?.basalDeliveryState else {
```

Comment:
This is inferring an enactment from a current state. The semantics of loopEnacted are different than basalDeliveryState. Guessing that can't be resolved in this ticket (i.e. we don't have enough info in StoredDosingDecision to actually know if a temp basal was enacted in this decision). If that is true, then please create Jira ticket covering that we need to address this mismatch, which will likely involve updating StoredDosingDecision to have a new field.

Reply:
Oh, I see you did now.. nevermind. :)